R/convert_na_to.R

```diff
@@ -120,16 +120,17 @@ convert_na_to.factor <- function(x, replacement = NULL, verbose = TRUE, ...) {
 #' @rdname convert_na_to
 #' @export
 convert_na_to.character <- function(x, replacement = NULL, verbose = TRUE, ...) {
-  if (is_empty_object(replacement) || !is.character(replacement)) {
+  if (is_empty_object(replacement) || !is.character(replacement) & !is.numeric(replacement)) {
     if (isTRUE(verbose)) {
```

Comment:
Should this be `&` or `&&`?

Reply:
I have read about the booleans && and || but I still do not understand them very well (is it just generally more efficient/faster?). In any case, I have changed it so.

Reply:
See `?'&'`:

> & and && indicate logical AND and | and || indicate logical OR. The shorter forms performs elementwise comparisons in much the same way as arithmetic operators. The longer forms evaluates left to right, proceeding only until the result is determined. **The longer form is appropriate for programming control-flow and typically preferred in *if* clauses.**


``` r
# element-wise logical combination
c(TRUE, TRUE, FALSE) | c(FALSE, TRUE, FALSE)
#> [1]  TRUE  TRUE FALSE

# element-wise logical combination
FALSE | c(FALSE, TRUE, FALSE)
#> [1] FALSE  TRUE FALSE

# warning, length >= 3
c(TRUE, TRUE, FALSE) || c(FALSE, TRUE, FALSE)
#> Warning in c(TRUE, TRUE, FALSE) || c(FALSE, TRUE, FALSE): 'length(x) = 3 > 1' in
#> coercion to 'logical(1)'
#> [1] TRUE

# LHS and RHS are *one* logical expression
any(c(TRUE, TRUE, FALSE)) || all(c(FALSE, TRUE, FALSE))
#> [1] TRUE
```

<sup>Created on 2022-08-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>